bliss/models/location_encoder.py

```diff
@@ -399,25 +365,20 @@ def training_step(self, batch, batch_idx, optimizer_idx=0):
         return loss
 
     def _get_loss(self, batch: Dict[str, Tensor]):
+        catalog_dict = {
+            "locs": batch["locs"][:, :, :, 0 : self.max_detections],
+            "log_fluxes": batch["log_fluxes"][:, :, :, 0 : self.max_detections],
+            "galaxy_bools": batch["galaxy_bools"][:, :, :, 0 : self.max_detections],
+            "n_sources": batch["n_sources"].clamp(max=self.max_detections),
+        }
+        true_catalog = TileCatalog(self.tile_slen, catalog_dict)
 
-        true_catalog = TileCatalog(
-            self.tile_slen,
-            {
-                "locs": batch["locs"][:, :, :, 0 : self.max_detections],
-                "log_fluxes": batch["log_fluxes"][:, :, :, 0 : self.max_detections],
-                "galaxy_bools": batch["galaxy_bools"][:, :, :, 0 : self.max_detections],
-                "n_sources": batch["n_sources"].clamp(max=self.max_detections),
```

Comment:
Is all the slicing/clamping necessary here? How would we have more than `self.max_detections` sources in the first place?

Reply:
For the m2 example, up to five sources can be generated per tile during sleep phase training. We may need to ask Bryan about this